src/documentation/sdks/python-type-safety.md

```diff
@@ -0,0 +1,138 @@
+---
+layout: default
+title: "Ensuring Python Type Safety"
+permalink: /documentation/sdks/python-type-safety/
+---
+#  Ensuring Python Type Safety
+
+Python is a dynamically-typed language with no static type checking. Because of the way Python's type checking works, as well as the deferred nature of runner execution, developer productivity can easily become bottle-necked by time spent investigating type-related errors.
+
+The Apache Beam SDK for Python uses **type hints** during pipeline construction and runtime to try to emulate the correctness guarantees achieved by true static typing. Additionally, using type hints lays some groundwork that allows the backend service to perform efficient type deduction and registration of `Coder` objects.
+
+Python version 3.5 introduces a module called **typing** to provide hints for type validators in the language. The Beam SDK for Python, based on Python version 2.7, implements a subset of ([PEP 484](https://www.python.org/dev/peps/pep-0484/)) and aims to follow it as closely as possible in its own typehints module.
+
+## Benefits of Type Hints
+
+The Beam SDK for Python includes some automatic type checking: for example, some `PTransform`s, such as `Create` and simple `ParDo` transforms, attempt to deduce their output type given their input. However, the Beam cannot infer types in all cases. Therefore, the recommendation is that you declare type hints to aid you in performing your own type checks if necessary.
+
+When you use type hints, the runner raises exceptions during pipeline construction time, rather than runtime. For example, the runner generates an exception if it detects that your pipeline applies mismatched `PTransforms` (where the expected outputs of one transform do not match the expected inputs of the following transform). These exceptions are raised whether your pipeline is executing locally or remotely. Introducing type hints for the `PTransform`s you define allows you to catch potential bugs up front in the local runner, rather than after minutes of execution into a deep, complex pipeline.
```

Comment:
Instead of `These exceptions are raised whether your pipeline is executing locally or remotely.`

maybe use `These exceptions are raised at pipeline construction time regardless of where your pipeline will execute.`